utils/auth.js

```diff
@@ -13,3 +13,23 @@ export function getUserFromCookie() {
     return null;
   }
 }
+
+
+export async function getUserFromRequest(req) {
+  const cookieHeader = req.headers.get('cookie');
+  if (!cookieHeader) return null;
+
+  const token = cookieHeader
```

Comment:
this looks hacky to get the token.
Consider using this https://nextjs.org/docs/app/api-reference/functions/cookies from next js